design/layer2-bind-interfaces.md

```diff
@@ -0,0 +1,56 @@
+# Configuring announcing interfaces
+
+## Summary
+
+The purpose of this enhancement is to make Speakers only announce VIPs on specified network interfaces.
+
+Previous discussions about this topic can be found in [this issue](https://github.com/metallb/metallb/issues/277).
+
+## Motivation
+
+A system may have multiple networks based on functions, and thus people want MetalLB only works on specified networks.
+In BGP mode, this situation can be handled by BGP configuration, while in L2 mode,  there is no way to achieve this purpose.
+
+[<img src="./images/example_system.png" width="800"/>](./images/example_system.png)
+
+### Goals
+- The network interfaces the MetalLB uses to broadcast VIP can be specified.
+
+### Non-Goals
+- Change any core functions of MetalLB
+
+## Data model
+
+Based on the current MetalLB CRD design (`8d906f0abef3d7ed5b0435fa4edfbe5e37b38557`), we'd like to add a new field `BindInterfaces` to `L2Advertisement`.
+
+`BindInterfaces` describes network interface names of the Nodes MetalLB should use. It is a list of interfaces' names, and it is optional. Once users set "bindInterface", then Speakers should only create announcers on these interfaces. Otherwise, Speakers keep the default behavior which creates announcers on every interface.
```

Comment:
```suggestion
`BindInterfaces` describes network interface names of the nodes MetalLB should use. It is a list of interfaces names, and it is optional. Once users set "bindInterface", then the Speakers should only create announcers on these interfaces. Otherwise, the Speakers keep the default behavior which creates announcers on every interface.
```